### PR TITLE
Add data migration to update legacy degree grade values 

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context

When entering a degree grade, in addition to allowing free text input we
also offer the candidate a selection of radio buttons mapped to one of
several common values. We've provided radio buttons historically,
however the actual stored database value for these standard options
changed when we added structured HESA data to degrees. Rather than
saving a short string such as `upper_second`, we now simply persist the
HESA description (`Upper second-class honours (2:1)`).

This prompted a change to how we display these grades on the frontend. Instead
of translating database values such as `upper_second` to a presentable
form via I18n, we now simply show the database value. This works fine
for anything with a standard HESA grade or a free-text input, but
doesn't look so great for older degree records that simply display
`upper_second` on the frontend.
## Changes proposed in this pull request

We can easily map those older values to the current HESA descriptions.
This commit adds a migration which applies that mapping and also adds
the corresponding HESA code to the degree record.

### Before migration
Screenshot_20200924_203241

### After migration
Screenshot_20200924_203538

## Link to Trello card
https://trello.com/c/kRTjP8aV

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Review by davidgisbey — APPROVED** [review]
lgtm :shipit:

File: db/migrate/20200924121000_migrate_old_degree_grade_values_to_hesa_descriptions.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class MigrateOldDegreeGradeValuesToHesaDescriptions < ActiveRecord::Migration[6.0]
+  def change
```
Comment (stevehook) [inline db/migrate/20200924121000_migrate_old_degree_grade_values_to_hesa_descriptions.rb:2]: As this is non-reversible (and doesn't need to be) would it be best to make this an `up` and have an empty `down` method?

**Review by tvararu — APPROVED** [review]

**Review by stevehook — APPROVED** [review]
Looks good. Minor comment inline.

File: db/migrate/20200924121000_migrate_old_degree_grade_values_to_hesa_descriptions.rb (hunk 1 above)
Comment (malcolmbaig) [reply db/migrate/20200924121000_migrate_old_degree_grade_values_to_hesa_descriptions.rb:2]: Doesn't trigger an exception on rollback which I think is probably fine?

**Pull request merged**